ogre2/src/Ogre2Scene.cc

```diff
@@ -1554,6 +1581,64 @@ bool Ogre2Scene::SkyEnabled() const
   return this->dataPtr->skyEnabled;
 }
 
+//////////////////////////////////////////////////
+bool Ogre2Scene::SetShadowTextureSize(LightType _lightType,
+    unsigned int _textureSize)
+{
+  // If _lightType is not supported, block with gzerr message
+  if (_lightType != LightType::DIRECTIONAL)
+  {
+    gzerr << "Light type [" << static_cast<int>(_lightType)
+          << "] is not supported." << std::endl;
+    return false;
+  }
+
+  // If _textureSize exceeds max possible tex size, then use default
+  if (_textureSize > this->dataPtr->maxTexSize / 2)
+  {
+    gzerr << "<texture_size> of '" << _textureSize
+          << "' exceeds maximum possible texture size,"
+          << " using default texture size" << std::endl;
+    return false;
+  }
+
+  // if _textureSize is an invalid texture size, then use default
+  if (_textureSize < 1024u || _textureSize > (this->dataPtr->maxTexSize / 2)
```

Comment:
I think it should be fine to set the lower limit to 512?

Reply:
ddde4d9a9686780b93296a1f79658540be698cb0